services/workflows-service/src/events/event-consumer.ts

```diff
@@ -1,21 +1,71 @@
-// import { WorkflowEventData } from '@/workflow/workflow-event-emitter.service';
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+
+import { WorkflowEventRawData } from '@/workflow/workflow-event-emitter.service';
 import { Injectable } from '@nestjs/common';
 import { OnEvent, EventEmitter2 } from '@nestjs/event-emitter';
 import { ListenerFn } from './types';
+import axios from 'axios';
+import { randomUUID } from 'crypto';
+import { sortBy } from 'lodash';
 
 @Injectable()
 export class EventConsumerListener {
   constructor(private eventEmitter: EventEmitter2) {}
 
-  subscribe(event: string, callback: ListenerFn) {
-    this.eventEmitter.addListener(event, eventData => {
-      callback(undefined, eventData);
+  @OnEvent('workflow.context.changed')
+  handleWorkflowEvent(data: WorkflowEventRawData) {
+    const oldDocuments = (data.runtimeData.context as any)['documents'] || [];
+    const newDocuments = data.context?.['documents'] || [];
+
+    const documentIdentifier = (doc: any) => {
+      return `${doc.category as string}$${doc.type as string}$${doc.issuer?.country as string}`;
```

Comment:
change to `${doc.category as string}-${doc.type as string}-${doc.issuer?.country as string}`.toLowerCAse()